### PR TITLE
[GEOS-8946]: Fix IllegalArgumentExcep on turboJPEG writing

### PR DESCRIPTION
**Pull request by dromagnoli**

### CODE REVIEW

**Comment by dromagnoli** [comment]
This PR depends on https://github.com/geotools/geotools/pull/2092

File: src/extension/libjpeg-turbo/src/test/java/org/geoserver/map/turbojpeg/TurboImageWorkerTest.java [hunk 1]
```diff
@@ -66,6 +72,44 @@ public void errors() throws IOException {
...
+        NoDataContainer noData = new NoDataContainer(noDataValue);
+        final IndexColorModel icm = new IndexColorModel(8, 256, bb, bb, bb);
+        final WritableRaster raster =
+                RasterFactory.createWritableRaster(icm.createCompatibleSampleModel(512, 512), null);
+        for (int i = raster.getMinX(); i < raster.getMinX() + raster.getWidth(); i++) {
+            for (int j = raster.getMinY(); j < raster.getMinY() + raster.getHeight(); j++) {
+                if (i - raster.getMinX() < raster.getWidth() / 2) {
+                    raster.setSample(i, j, 0, (i + j) / 32);
+                } else {
+                    raster.setSample(i, j, 0, 200);
```
Comment (aaime) [inline src/extension/libjpeg-turbo/src/test/java/org/geoserver/map/turbojpeg/TurboImageWorkerTest.java]: So the color expansion would create alpha out of the nodata and the boom, right?

File: src/extension/libjpeg-turbo/src/test/java/org/geoserver/map/turbojpeg/TurboImageWorkerTest.java [hunk 2]
```diff
@@ -66,6 +72,44 @@ public void errors() throws IOException {
...
+                }
+            }
+        }
+        // Set no data
+        BufferedImage bi = new BufferedImage(icm, raster, false, null);
+        PlanarImage planarImage = PlanarImage.wrapRenderedImage(bi);
+        planarImage.setProperty(NoDataContainer.GC_NODATA, noData);
```
Comment (dromagnoli) [inline src/extension/libjpeg-turbo/src/test/java/org/geoserver/map/turbojpeg/TurboImageWorkerTest.java]: Yep. here it set the nodata property which is taken by the ImageWorker to setup the Alpha:
https://github.com/geotools/geotools/blob/master/modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java#L1771
So without the fix, it was throwing the IllegalArgumentException

File: src/extension/libjpeg-turbo/src/test/java/org/geoserver/map/turbojpeg/TurboImageWorkerTest.java (hunk 1 above)
Comment (dromagnoli) [reply src/extension/libjpeg-turbo/src/test/java/org/geoserver/map/turbojpeg/TurboImageWorkerTest.java]: See my comment below.

**Comment by aaime** [comment]
GeoTools PR merged, this needs to be re-built once the build server has pushed the new gt jars in the repo

**Comment by bradh** [comment]
Will rekick CI servers.

**Comment by aaime** [comment]
Merged. Did not touch the ticket, up to you, also depending on whether you want to backport or not

**Pull request merged**